app/scripts/controllers/transactions/index.js

```diff
@@ -1345,8 +1345,9 @@ export default class TransactionController extends EventEmitter {
    *
    * @param {number} txId - the tx's Id
    * @param {string} actionId - actionId passed from UI
+   * @param isApprovalAcceptanceNeeded
    */
-  async approveTransaction(txId, actionId) {
+  async approveTransaction(txId, actionId, isApprovalAcceptanceNeeded = true) {
```

Comment:
Given the complexity of this controller, is it best to put this in an optional options object so it's explicit what the boolean is doing when we call this? Also perhaps cleaner not to couple the argument name to the outcome so a call could look something like:

```
approveTransaction(someId, someActionId, { hasApprovalRequest: false });
```

Reply:
Sure thing, that might improve readability as well

Reply:
Fixed